### ISSUE TITLE
Gotype should  check an entire package instead of one file.

### ISSUE THREAD

**Issue by DmitriyMV** [go-lang-plugin-org/go-lang-idea-plugin#1990]
Right now, if your project has more than one file and plugin prduces errors. That's because currently gotype is only checking selected file without getting types from others, so it doesn't know anything about them. Solution is quite simple - just check the folder and not the file.

In case of `/src/github.com/username/goproject/main.go` should become `/src/github.com/username/goproject`

**Comment by dlsniper** [comment]
That's actually incorrect in my case. I've tested the integration just now and in my example this won't work.

I've checked a file that's in a deeper package structure and I get an error about missing imports (even if the files, packages, are on disk). I've tried using a directory instead of a file and I've got the same issue.

At the moment I don't have a proper solution for this to work but as soon as I have one I'll add it here.

**Comment by dlsniper** [comment]
Ok so I've found the root of the issue.
According to https://github.com/golang/go/issues/10969 which points to https://github.com/golang/go/issues/10249#issuecomment-86671707 this happens because `go install` isn't used on the package at least once before applying `gotype`. I'll try to submit a fix for this soonish.

**Comment by zolotov** [comment]
> Solution is quite simple - just check the folder and not the file.

So select the folder and run gotype action, what's the problem?

**Comment by dlsniper** [comment]
> So select the folder and run gotype action, what's the problem?

When using import statements for other packages that won't work, see my comment.

**Comment by zolotov** [comment]
The original issue is about running on directory.

As for `go install` and other stuff. Since `gotype` is not really smart, I prefer do not fix it and wait until we'll get our own type checking system.

**Comment by dlsniper** [comment]
> The original issue is about running on directory.

Nope, I believe that the author of the issue didn't had the full understanding of the issue that's why it was suggested to run `gotype` on a folder.

> I prefer do not fix it and wait until we'll get our own type checking system.

Sure but the feature is already there so then maybe it should be removed as it doesn't work in most of the use-cases (I rarely work on packages that don't import other packages so for me this is broken almost always.

**Comment by DmitriyMV** [comment]
No - gotype just requires that imported packages should have object files (the ones in /pkg dir). Anyway, I think that the issue can be closed, because @zolotov answer (choosing directory instead of file) is working.

**Comment by dlsniper** [comment]
> No - gotype just requires that imported packages should have object files (the ones in /pkg dir).

Which in my case doesn't happen as I rarely (almost never) use `go install` on a package. Also, if we move more and more functionality to the IDE (which we want in the first place), there are less chances for packages to be "installed" thus the need to have `go install`.

**Comment by DmitriyMV** [comment]
But you are using `go get` functionality to install 3rd party libraries. gotype is 3rd party analyzing tool, so it contains some requirements for its correct usage. Of course I already mentioned that with so many 3rd party tools in Go ecosystem, a more general approach to this problem would be better.

**Comment by dlsniper** [comment]
I meant on packages that I newly create, I'm using go get to fetch even non-go repos nowadays but I almost never use go get on my own packages... Also as the IDE runs projects with go build or go run it further shows either the need to run go install when executing tools such as these or a different approach to running apps (run go install first there?). 

> gotype is 3rd party analyzing tool

No, it's meant to be used with any go package, not just 3rd party.

**Comment by DmitriyMV** [comment]
No - I mean't it is not one of the "core go tool's" like `gofmt` or `go doc`.

Running `go install` would be nice, I think.
